image/tarexport/save_oci.go

```diff
@@ -0,0 +1,419 @@
+package tarexport
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/image"
+	"github.com/docker/docker/layer"
+	"github.com/docker/docker/pkg/archive"
+	"github.com/docker/docker/pkg/system"
+	"github.com/opencontainers/go-digest"
+	imgspec "github.com/opencontainers/image-spec/specs-go"
+	ociv1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+)
+
+type layerInfo struct {
+	digest digest.Digest
+	size   int64
+}
+
+type ociSaveSession struct {
+	*tarexporter
+	images       map[image.ID]*imageDescriptor
+	name         string
+	savedImages  map[image.ID]ociv1.Manifest // cache image.ID -> manifest bytes
+	diffIDsCache map[layer.DiffID]*layerInfo
+	outDir       string
+	index        ociv1.Index
+}
+
+func (l *tarexporter) getRefs() (map[string]string, map[string]reference.NamedTagged, error) {
```

Comment:
This method really has nothing to do with `tarexporter`, and with my proposed change of removing `Options` from tarexporter this will need to move somewhere else.

I think this can be split into two functions:

```go
func normalizeOCIRefs(refs map[string]string) error {
   ...
}

func refsToTaggedRefs(refs map[string]string) (map[string]reference.NamedTagged, error) {
   ...
}
```

`parseOCINames` can call `normalizeOCIRefs` since it doesn't need the tagged, and load can call `refsToTaggedRefs`.